cmd/build_test.go

```diff
@@ -970,13 +977,602 @@ func TestBuildBundleModeIgnoreFlag(t *testing.T) {
+		expManifest string
+		expErrs     []string
+	}{
+		{
+			note: "v0 bundle rego-version",
+			files: map[string]string{
+				".manifest": `{"rego_version": 0}`,
+				"test.rego": `package test
+
+p[42] {
+	input.x == 1
+}`,
+			},
+			expManifest: `{"revision":"","roots":[""],"rego_version":0}`,
+		},
+		{
+			note: "v1 bundle rego-version",
+			files: map[string]string{
+				".manifest": `{"rego_version": 1}`,
+				"test.rego": `package test
+
+p contains 42 if {
+	input.x == 1
+}`,
+			},
+			expManifest: `{"revision":"","roots":[""],"rego_version":1}`,
+		},
+		{
+			note: "v0 bundle rego-version, v1 per-file override",
+			files: map[string]string{
+				".manifest": `{
+	"rego_version": 0,
+	"file_rego_versions": {
+		"*/test2.rego": 1
+	}
+}`,
+				"test1.rego": `package test
+
+p[1] {
+	input.x == 1
+}`,
+				"test2.rego": `package test
+
+p contains 2 if {
+	input.x == 1
+}`,
+			},
+			expManifest: `{"revision":"","roots":[""],"rego_version":0,"file_rego_versions":{"%ROOT%/test2.rego":1}}`,
+		},
+		{
+			note: "v0 bundle rego-version, v1 per-file override, missing v1 keywords in v1 file",
+			files: map[string]string{
+				".manifest": `{
+	"rego_version": 0,
+	"file_rego_versions": {
+		"*/test2.rego": 1
+	}
+}`,
+				"test1.rego": `package test
+
+p[1] {
+	input.x == 1
+}`,
+				"test2.rego": `package test
+
+p[2] {
+	input.x == 1
+}`,
+			},
+			expErrs: []string{
+				"rego_parse_error: `if` keyword is required before rule body",
+				"rego_parse_error: `contains` keyword is required for partial set rules",
+			},
+		},
+		{
+			note: "v0 bundle rego-version, v1 per-file override, v1 keywords but no v1 imports in v0 file",
+			files: map[string]string{
+				".manifest": `{
+	"rego_version": 0,
+	"file_rego_versions": {
+		"*/test2.rego": 1
+	}
+}`,
+				"test1.rego": `package test
+
+p contains 1 if {
+	input.x == 1
+}`,
+				"test2.rego": `package test
+
+p contains 2 if {
+	input.x == 1
+}`,
+			},
+			expErrs: []string{
+				"rego_parse_error: var cannot be used for rule name",
+				"rego_parse_error: number cannot be used for rule name",
+			},
+		},
+		{
+			note:  "multiple bundles with different rego-versions",
+			roots: []string{"bundle1", "bundle2"},
+			files: map[string]string{
+				"bundle1/.manifest": `{
+	"roots": ["test1"],
+	"rego_version": 0,
+	"file_rego_versions": {
+		"*/test2.rego": 1
+	}
+}`,
+				"bundle1/test1.rego": `package test1
+p[1] {
+	input.x == 1
+}`,
+				"bundle1/test2.rego": `package test1
+p contains 2 if {
+	input.x == 1
+}`,
+				"bundle2/.manifest": `{
+	"roots": ["test2"],
+	"rego_version": 1,
+	"file_rego_versions": {
+		"*/test4.rego": 0
+	}
+}`,
+				"bundle2/test3.rego": `package test2
+p contains 3 if {
+	input.x == 1
+}`,
+				"bundle2/test4.rego": `package test2
+p[4] {
+	input.x == 1
+}`,
+			},
+			expManifest: `{"revision":"","roots":["test1","test2"],"rego_version":0,"file_rego_versions":{"%ROOT%/bundle1/test2.rego":1,"%ROOT%/bundle2/test3.rego":1}}`,
```

Comment:
Correct; since the system-wide default is v0.

We don't include per-file overrides when the version matches the bundle-global version to avoid unnecessary bloat in the manifest, should the bundle contain a lot of Rego files. A side benefit is that if all file versions matches the global version (the common case), the `file_rego_versions` manifest entry will be empty (and actually not written at all).